### PR TITLE
Lesson 4.1: Performance TypeScript Testing Project Structure

### PR DESCRIPTION
**Pull request by nihalerdal**
Here is my lesson_4_1 pull request for the review.

### CODE REVIEW

File: src/Components/InputWithLabel.jsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React from "react";
 import styles from "./InputWithLabel.module.css";
+import PropTypes from "prop-types";
 
 function InputWithLabel(props) {
```
Comment (AnnaPestova1) [inline src/Components/InputWithLabel.jsx]: just interesting - why you did not deconstruct props in this component?

**Review by AnnaPestova1 — APPROVED** [review]
Great job Nihal! 
You can merge

File: src/Components/TodoListItem.jsx [hunk 2]
```diff
@@ -19,4 +20,10 @@ function TodoListItem({ title, onRemoveTodo, id }) {
   );
 }
 
+TodoListItem.propTypes = {
+  title: PropTypes.string.isRequired,
+  onRemoveTodo: PropTypes.func,
+  id: PropTypes.string.isRequired,
```
Comment (AnnaPestova1) [inline src/Components/TodoListItem.jsx:26]: isRequired is a very useful flag: you can provide it for all props that cannot be empty

File: src/Components/InputWithLabel.jsx (hunk 1 above)
Comment (nihalerdal) [reply src/Components/InputWithLabel.jsx]: just forgot:) done now✅

Comment (nihalerdal) [reply src/Components/InputWithLabel.jsx]: Also added stretch goal, TodoContainer.jsx

File: src/Components/TodoContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
+import React from "react";
```
Comment (AnnaPestova1) [inline src/Components/TodoContainer.jsx:1]: Great job!

**Pull request merged**